components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Engine/StandaloneJobOrchestrator.cs

```diff
@@ -786,82 +786,100 @@ IJsonSerializer jsonSerializer
+                                    found = true;
+                                }
                             }
+                        }
 
+                        if (!found) {
+                            var entryJobId = _publishedNodesJobConverter.
+                                ToConnectionModel(request, _standaloneCliModel).CreateConnectionId();
                             foreach (var assignedJob in _assignedJobs) {
-                                if (newJob.Job.Id == assignedJob.Value.Job.Id) {
-                                    _assignedJobs[assignedJob.Key] = newJob;
-                                    found = true;
+                                if (entryJobId == assignedJob.Value.Job.Id) {
+                                    found = _assignedJobs.Remove(assignedJob.Key, out _);
+                                    _publisherDiagnosticInfo.Remove(assignedJob.Value.Job.Id, out _);
                                     break;
                                 }
                             }
-                            if (!found && _availableJobs.ContainsKey(newJob.Job.Id)) {
-                                _availableJobs[newJob.Job.Id] = newJob;
-                                found = true;
+                            if (!found) {
+                                found = _availableJobs.Remove(entryJobId, out _);
                             }
                         }
                     }
+                    finally {
+                        _lockJobs.Release();
+                    }
 
                     if (!found) {
-                        var entryJobId = _publishedNodesJobConverter.
-                            ToConnectionModel(request, _standaloneCliModel).CreateConnectionId();
-                        foreach (var assignedJob in _assignedJobs) {
-                            if (entryJobId == assignedJob.Value.Job.Id) {
-                                found = _assignedJobs.Remove(assignedJob.Key, out _);
-                                _publisherDiagnosticInfo.Remove(assignedJob.Value.Job.Id, out _);
-                                break;
-                            }
-                        }
-                        if (!found) {
-                            found = _availableJobs.Remove(entryJobId, out _);
-                        }
+                        throw new MethodCallStatusException((int)HttpStatusCode.NotFound,
+                            $"Endpoint not found: {request.EndpointUrl}");
                     }
                 }
-                finally {
-                    _lockJobs.Release();
-                }
+                else {
 
-                if (!found) {
-                    throw new MethodCallStatusException((int)HttpStatusCode.NotFound,
-                        $"Endpoint not found: {request.EndpointUrl}");
+                    // Remove all entries
+                    _publishedNodesEntries.Clear();
```

Comment:
We should also update `_lastKnownFileHash ` like this together with `_publishedNodesEntries.Clear();`:

```
_publishedNodesEntries.Clear();
_lastKnownFileHash = string.Empty;
```

Reply:
`PersistPublishedNodes()` does this already.